main.go

```diff
@@ -201,6 +203,22 @@ func Exec(ctx context.Context, args Args) error {
 	if err := runCommand(cmdArgs); err != nil {
 		logrus.Fatalln("error executing jfrog rt build-publish command:", err)
 	}
+	
+	// Add Principal directly via REST API using build trigger
+	buildTrigger := os.Getenv("DRONE_BUILD_TRIGGER")
+	if buildTrigger != "" {
+		logrus.WithFields(logrus.Fields{
+			"principal": buildTrigger,
+		}).Info("Adding Principal information via REST API")
+		
+		// Brief pause for build-publish to complete and be available via API
+		time.Sleep(5 * time.Second)
```

Comment:
instead of fixed 5 seconds sleep poll the API with a short timeout/ctx until the build record appears